packages/compile-solidity/compilerSupplier/loadingStrategies/VersionRange.js

```diff
@@ -110,12 +158,12 @@ class VersionRange extends LoadingStrategy {
     try {
```

Comment:
This appears to be another function using its own retry logic, for which I'd once again suggest using `axios-retry` instead.

Reply:
Also, I know this is preexisting, but the name of this function is confusing.  I'd suggest renaming it to `getSolcByUrlAndCacheIt` to make it clearer.  (As it is, it sounds like it takes a URL and a cache as input and returns a solc as output, when in fact it takes just a url as input, returns a solc as output, and also caches that solc.)

Reply:
> This appears to be another function using its own retry logic, for which I'd once again suggest using `axios-retry` instead.

Opened [issue](https://github.com/trufflesuite/truffle/issues/4261) for this

Reply:
> Also, I know this is preexisting, but the name of this function is confusing.

Renamed it to `getAndCacheSolcByUrl`